### PR TITLE
[Wiki] Update Fetch latest updates section and fix broken links

### PR DESCRIPTION
**Pull request by eriqua**
# Description

>Thank you for your contribution !

> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: docs/wiki/Fetching latest changes - Scenario 1 Consume library.md [hunk 1]
```diff
@@ -16,33 +16,23 @@ The update process is the following:
...
 
-In this scenario every time you'll fetch updates from the public CARML repository merge conflicts are expected. You'll have to compare the new public code with your customized one and re-apply your customizations to the updated code.
+If your copy deviates from the upstream version due to customizations you applied to the code, you'll have to re-apply your customizations to the updated code. This process may be automated, by script or CI, if customization tasks are repeatable.
 
-This process can be automated, by script or CI, if customization tasks are repeatable.
+> **Note**: If customizations are general improvements which may be useful for the public, the recommendation is to [contribute](./Contribution%20guide) to the public CARML repository so that your updates can improve the public library. This way, your changes will be already available the next time you will fetch from upstream, modules will be already tested with your changes and you will not need to take care of customization on each update.
```
Comment (ahmadabdalla) [inline docs/wiki/Fetching latest changes - Scenario 1 Consume library.md:33]: Consider rewording this bit.. maybe something like: 

"your changes will already be available the next time you fetch from upstream, as modules would already been tested, and would not conflict with your customizations."

File: docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md [hunk 2]
```diff
@@ -48,42 +52,32 @@ Clone/download CARML repository to create a local copy of it, as explained in Az
...
-# 7. Push updated code
+# 7. Update module test files
 
-Push the updated local code to your remote public fork (for GitHub public repository scenario) or on your remote target repository for GitHub private repository and Azure DevOps private git scenarios.
+Follow the [Update module test files](./Getting%20started%20-%20Scenario%202%20Onboard%20module%20library%20and%20CI%20environment#5-update-module-test-files) procedure
 
 # 8. Test and publish modules
 
-If actions/ pipelines are enabled test and publishing of modules starts automatically. In some cases triggers for the pipelines are not enabled, in these cases start actions/pipelines to run tests and publish modules.
+Push the updated local code to your remote repository. If actions are enabled test and publishing of modules will start automatically.
```
Comment (ahmadabdalla) [inline docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md:83]: If actions are enabled,

File: docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md [hunk 3]
```diff
@@ -48,42 +52,32 @@ Clone/download CARML repository to create a local copy of it, as explained in Az
...
+The backup folder from step 1, can be used to compare your local copy with your synced copy coming from the latest version. For example, the `'Compare selected'` [function](https://vscode.one/diff-vscode/) in Visual Studio Code can be leveraged for that purpose.
 
-# 5. Update module module test files
+If your copy deviates from the upstream version due to customizations you applied to the code, you'll have to re-apply those customizations to the updated code. This process may be automated, by script or CI, if customization tasks are repeatable.
 
-Follow the [Update module module test files](./Getting%20started%20-%20Scenario%202%20Onboard%20module%20library%20and%20CI%20environment#5-update-module-module-test-files) procedure
+> **Note**: If customizations are general improvements which may be useful for the public, the recommendation is to [contribute](./Contribution%20guide) to the public CARML repository so that your updates can improve the public library. This way, your changes will be already available the next time you will fetch from upstream, modules will be already tested with your changes and you will not need to take care of customization on each update.
```
Comment (ahmadabdalla) [inline docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md:71]: Same as other rewording comment

File: docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md (hunk 2 above)
Comment (eriqua) [reply docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md]: done

File: docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md (hunk 3 above)
Comment (eriqua) [reply docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md]: done

File: docs/wiki/Fetching latest changes - Scenario 1 Consume library.md (hunk 1 above)
Comment (eriqua) [reply docs/wiki/Fetching latest changes - Scenario 1 Consume library.md]: done

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**